### ISSUE TITLE
unexpected "Error: Unsupported attribute" for type "any" variable import (0.12.29/0.13.0)

### ISSUE THREAD

**Issue by tony-kerz** [hashicorp/terraform#25888]
### Terraform Version

```
v0.12.29 and v0.13.0
```

### Terraform Configuration Files

__variables file__
```
variable "rt53" {
  type    = any
  default = {}
}
```

__tfvars file__
```terraform
rt53 = {
  local = {
    private_zone_name = null
  }
}
```

__tf file__
```
locals {
  rt53 = {
    is_local = (var.rt53.local.private_zone_name == null) ? 0 : 1
  }
}
```

### Expected Behavior

terraform should resolve variables similarly in `import` as in `plan`

### Actual Behavior

terraform `init` and `plan` work successfully, but `import` yields the following:

```
terraform import -var-file=tfvars/lab.tfvars module.vpc.aws_vpc.this vpc-04d88d30de2b4c0ad
```
```
Error: Unsupported attribute

  on module/vpc/rt53.tf line 3, in locals:
   3:     is_local = (var.rt53.local.private_zone_name == null) ? 0 : 1
    |----------------
    | var.rt53 is object with no attributes

This object does not have an attribute named "local".
```

### Steps to Reproduce

i could not easily reproduce in a small application, so there must be some nuance with my larger project.

i did notice that adding the variable values as the `default` in the module's `variables.tf` file brute-forced them into the context and it side-stepped the errors...

### Additional Context

the item i was trying to import was not directly related to the item mentioned in the errors

### References

seems similar to #16139

**Comment by alisdair** [comment]
@tony-kerz Thanks for reporting this! I think I've been able to reproduce the behaviour you described [in this minimal configuration](https://github.com/danieldreier/terraform-issue-reproductions/tree/master/25888). Could you take a look at this and see if it lines up with your larger project?

**Comment by alisdair** [comment]
After a quick investigation I'm unable to explain what's happening here. But I was able to find that removing the `default` value for the variable declared in the module allows the import to work successfully. Noting this here in the hope that it's a useful hint for whoever investigates this next.

This also might be useful as a workaround for your configuration, @tony-kerz—especially as the existing default value of `{}` seems to be invalid for how the configuration uses the variable.

**Comment by alisdair** [comment]
I think I have a fix for this in #25890.

**Comment by tony-kerz** [comment]
thanks @alisdair, verifying that removing the default allows the passed variable into context for import and things work as expected! 

this is a fine work-around for me in this case, but of course, it's great that you might have a fix, that kind of turn-around doesn't happen every day, cheers! 🍺
